### PR TITLE
Allocator master flags

### PR DESCRIPTION
**Pull request by Lqp1**

### CODE REVIEW

File: src/master/main.cpp [hunk 1]
```diff
@@ -342,9 +343,16 @@ int main(int argc, char** argv)
...
 
-  CHECK_NOTNULL(allocator.get());
+  Allocator * alloc = allocator.get();
+  CHECK_NOTNULL(alloc);
   LOG(INFO) << "Using '" << flags.allocator << "' allocator";
 
+  if (flags.allocator == "hierarchical")
+  {
+      HierarchicalAllocatorProcess *a = (HierarchicalAllocatorProcess*) alloc;
+      a->setParams(flags.hierarchical_recovery_factor, flags.hierarchical_recovery_timeout);
```
Comment (Lqp1) [inline src/master/main.cpp]: I'm still on here to make it a little more OOP compliant

File: src/master/constants.hpp [hunk 2]
```diff
@@ -48,6 +48,9 @@ constexpr double MIN_CPUS = 0.01;
 // Minimum amount of memory per offer.
 constexpr Bytes MIN_MEM = Megabytes(32);
 
+constexpr Duration DEFAULT_RECOVERY_TIMEOUT = Minutes(10);
+constexpr float DEFAULT_AGENT_RECOVERY_FACTOR = 0.95;
```
Comment (kamaradclimber) [inline src/master/constants.hpp]: if we submit this upstream, we'll need to update the default value. I would suggest to set it to 0.8 (current default in mesos)

**Review by kamaradclimber — DISMISSED** [review]

**Review by kamaradclimber — APPROVED** [review]

File: include/mesos/allocator/allocator.hpp [hunk 3]
```diff
@@ -67,6 +67,10 @@ struct Options
   bool publishPerFrameworkMetrics = true;
 
   Option<std::string> slaveSorterResourceWeights;
+
+  // Recovery options
+  Duration recoveryTimeout = Minutes(10);
+  float agentRecoveryFactor = 0.80;
```
Comment (komuta) [inline include/mesos/allocator/allocator.hpp:73]: why not re-use the DEFAULT_* constants?

Comment (Lqp1) [reply include/mesos/allocator/allocator.hpp:73]: You're right, I'll fix this in upstream PR
Those values will never be used as they are overriden by flags (whose default values are DEFAULT_* constants)
I can also remove affectation here.

**Comment by Lqp1** [comment]
https://github.com/apache/mesos/pull/367

**Pull request merged**